app/code/Magento/QuoteGraphQl/Model/Cart/AddSimpleProductToCart.php

```diff
@@ -126,20 +125,51 @@ private function extractQty(array $cartItemData): float
     /**
      * Extract Customizable Options from cart item data
      *
+     * @param \Magento\Catalog\Model\Product $product
      * @param array $cartItemData
      * @return array
+     * @throws GraphQlInputException
      */
-    private function extractCustomizableOptions(array $cartItemData): array
+    private function extractCustomizableOptions(\Magento\Catalog\Model\Product $product, array $cartItemData): array
     {
         $customizableOptions = $this->arrayManager->get('customizable_options', $cartItemData, []);
-
+        $optionTypeDetails = $this->getOptionTypeDetails($product);
         $customizableOptionsData = [];
-        foreach ($customizableOptions as $customizableOption) {
-            $customizableOptionsData[$customizableOption['id']] = $customizableOption['value'];
+
+        foreach ($customizableOptions as $optionItem) {
+            if (!isset($optionItem['value'])) {
+                throw new GraphQlInputException(__('Value is required for Option Id %1', $optionItem['id']));
+            }
+
+            $optId = $optionItem['id'];
+            if (in_array($optionTypeDetails[$optId], ["checkbox", "multiple"])) {
+                $customizableOptionsData[$optionItem['id']][] = $optionItem['value'];
```

Comment:
Looks like we can use `array_column` function

Reply:
$a = Array
(
    [0] => Array
        (
            [id] => 1
            [value] => 2
        )

    [1] => Array
        (
            [id] => 1
            [value] => 1
        )

)

$b = array_column($a, 'value', 'id'); returns

Array
(
    [1] => 1
)

array_column is not listed as multiple arrays when id is same
https://ideone.com/dY9TFt